addb2/sit.c

```diff
@@ -198,17 +198,30 @@ M0_INTERNAL int m0_addb2_storage_header(struct m0_stob *stob,
 static int it_init(struct m0_addb2_sit *it,
 		   struct m0_addb2_frame_header *h, m0_bindex_t start)
 {
-	struct m0_addb2_frame_header prev;
+	struct m0_addb2_frame_header header;
+	uint64_t                     last_frame_end;
 	int                          result;
 
 	if (start != 0) {
 		result = header_read(it, h, start);
 	} else {
+		/* Search for the last frame on the stob. */
 		while (1) {
-			result = header_read(it, &prev, h->he_prev_offset);
-			if (result != 0 || prev.he_seqno != h->he_seqno - 1)
+			result = header_read(it, &header, h->he_offset + h->he_size);
```

Comment:
This is search forward. How is the "rotation" handled correctly?

Reply:
search forward is not used to detect/handle "rotation". It's used for detection of last frame on the stob. Because in some cases we have in stob header copy of header of non-last frame. We must detect realy last frame on the stob because we use offset and size to detect the rotation below at the code.